### PR TITLE
Update dependencies

### PR DESCRIPTION
**Pull request by paolobarbolini**
I'm updating dependencies for another project and this came up as having many outdated dependencies. Unfortunately I don't have a Yubikey, nor does this project seem to have any tests, so I wasn't able to check for any regressions that this might have introduced.

Can 0.11 be released after this is merged?

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -20,14 +20,13 @@ path = "src/lib.rs"
...
 futures = { version = "0.3", optional = true }
-hmac = "0.10"
+hmac = "0.12"
 rand = "0.8"
 reqwest = { version = "0.11", features = ["blocking"] }
-sha-1 = "0.9"
+sha1 = "0.10"
 threadpool = "1.7"
-url = "1.7"
+form_urlencoded = "1"
```
Comment (pierre-l) [inline Cargo.toml:29]: What's the motivation behind this dependency change?

Comment (paolobarbolini) [reply Cargo.toml:29]: `url` v2 doesn't re-export `percent_encoding` anymore. I could have replaced `url` with `percent_encoding`, but seeing how all it was used for was to encode query strings I decided to go with `form_urlencoded` which is made specifically to deal with query strings.

All three crates are made by the same people, as you can see from the repo https://github.com/servo/rust-url

**Comment by pierre-l** [comment]
LGTM, thank you for this contribution.

I tested the happy path, no surprise.
Regarding test automation, we could probably add a unit test here and there but the best value would come from an integration test and it would require an OTP generator.

I'll let @wisespace-io review too & release as I don't have the release credentials anyway ;)

**Comment by paolobarbolini** [comment]
Awesome. I hope to get a Yubikey soon so that I'll too be able to develop things with them, and maybe contribute more to this crate 😃

**Comment by wisespace-io** [comment]
@pierre-l Do you have an account in crates.io? I can add you there as an owner. So you can do releases

**Comment by pierre-l** [comment]
@wisespace-io I just created one as `pierre-l`. I never released anything on crates.io yet, but I'll be happy to do so!

**Pull request merged**